numpyro/distributions/mixtures.py

```diff
@@ -211,3 +223,208 @@ def log_prob(self, value, intermediates=None):
         component_log_probs = self.component_distribution.log_prob(value)
         sum_log_probs = self.mixing_distribution.logits + component_log_probs
         return jax.nn.logsumexp(sum_log_probs, axis=-1)
+
+
+class MixtureGeneral(Distribution):
+    def __init__(
+        self, mixing_distribution, distributions, *, validate_args=None
+    ):
+        _check_mixing_distribution(mixing_distribution)
+
+        self._mixture_size = jnp.shape(mixing_distribution.probs)[-1]
+        if isinstance(distributions, Distribution):
```

Comment:
Hmm, do we need to support mixtures of the same distributions?

Reply:
Fair - we handle this use case in `Mixture`.